src/ctapipe/core/tool.py

```diff
@@ -429,46 +430,50 @@ def run(self, argv=None, raises=False):
                 self.log.info("Finished: %s", self.name)
                 Provenance().finish_activity(activity_name=self.name)
             except (ToolConfigurationError, TraitError) as err:
+                current_exception = err
                 self.log.error("%s", err)
                 self.log.error("Use --help for more info")
                 exit_status = 2  # wrong cmd line parameter
                 Provenance().finish_activity(
                     activity_name=self.name, status="error", exit_code=exit_status
                 )
-                if raises:
-                    raise
             except KeyboardInterrupt:
                 self.log.warning("WAS INTERRUPTED BY CTRL-C")
                 exit_status = 130  # Script terminated by Control-C
                 Provenance().finish_activity(
                     activity_name=self.name, status="interrupted", exit_code=exit_status
                 )
             except Exception as err:
-                self.log.exception("Caught unexpected exception: %s", err)
-                exit_status = 1  # any other error
+                current_exception = err
+                exit_status = getattr(err, "exit_code", 1)
+                if exit_status == 1:
+                    self.log.exception("Caught unexpected exception: %s", err)
+                else:
+                    self.log.error("Caught exception: %s", err)
```

Comment:
Why hide the traceback here?

Reply:
And even if it is an exception with an exit code, wouldn't it always be "unexpected" by definition?

Reply:
well... I actually have a regular case of "expected" exceptions... Not that it is a nice thing to have, but e.g. GDAS server is not responsive from time to time on a regular basis. However, since this is just a message in the log, and exception is actually known as will be reflected in its message, I will factor this out.

Reply:
Wouldn't this better be solved by something like exponential back-off inside the script?

E.g. https://pypi.org/project/retrying/

Reply:
No, as the server is sometimes unresponsive for a long-ish time and since it is not very critical, we just proceed differently.